samples/SampleApp/Startup.cs

```diff
@@ -105,10 +106,23 @@ public static Task Main(string[] args)
                         listenOptions.UseHttps(StoreName.My, "localhost", allowInvalid: true);
                     });
 
+                    options.ListenAnyIP(basePort + 5, listenOptions =>
+                    {
+                        listenOptions.UseHttps(httpsOptions =>
+                        {
+                            var localhostCert = CertificateLoader.LoadFromStoreCert("localhost", "My", StoreLocation.CurrentUser, allowInvalid: true);
+                            httpsOptions.ServerCertificateSelector = (features, name) =>
```

Comment:
I'm not sure I like this just being a settable property. What about adding a RegisterCertificateSelector method that takes a state object as well?

Reply:
Offline note: this is per endpoint, not per connection.